When a coroutine immediately ends, 'or_()' will crash

```python
def test_or_2():
    import asynckivy as ak
    async def _test():
        async def do_nothing():
            pass
        tasks = await ak.or_(ak.sleep_forever(), do_nothing())
        assert not tasks[0].done
        assert tasks[1].done
        nonlocal done; done = True
    done = False
    ak.start(_test())
    assert done
```

```
================================= FAILURES ==================================
_________________________________ test_or_2 _________________________________

    def test_or_2():
        import asynckivy as ak
        async def _test():
            async def do_nothing():
                pass
            tasks = await ak.or_(ak.sleep_forever(), do_nothing())
            assert not tasks[0].done
            assert tasks[1].done
            nonlocal done; done = True
        done = False
>       ak.start(_test())

tests/test_core.py:93: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
asynckivy/_core.py:17: in start
    coro.send(None)(step_coro)
tests/test_core.py:88: in _test
    tasks = await ak.or_(ak.sleep_forever(), do_nothing())
asynckivy/_core.py:60: in or_
    return await gather(coros, n=1)
asynckivy/_core.py:49: in gather
    start(task._run())
asynckivy/_core.py:17: in start
    coro.send(None)(step_coro)
asynckivy/_core.py:33: in _run
    self.done_callback()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def done_callback():
        nonlocal n_coros_left
        n_coros_left -= 1
        if n_coros_left == 0:
>           step_coro()
E           TypeError: 'NoneType' object is not callable
```

--- a/asynckivy/_core.py
+++ b/asynckivy/_core.py
@@ -37,8 +37,9 @@ class Task:
 def gather(coros:typing.Iterable[typing.Coroutine], *, n:int=None) -> typing.Sequence[Task]:
     coros = tuple(coros)
     n_coros_left = n if n is not None else len(coros)
-    step_coro = None
 
+    def step_coro(*args, **kwargs):
+        nonlocal n_coros_left; n_coros_left -= 1
     def done_callback():
         nonlocal n_coros_left
         n_coros_left -= 1
@@ -47,6 +48,9 @@ def gather(coros:typing.Iterable[typing.Coroutine], *, n:int=None) -> typing.Seq
     tasks = tuple(Task(coro, done_callback=done_callback) for coro in coros)
     for task in tasks:
         start(task._run())
+
+    if n_coros_left <= 0:
+        return tasks
 
     def callback(step_coro_):
         nonlocal step_coro

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -64,40 +64,77 @@ def test_gather2(ed_cls):
     assert done
 
 
-def test_or_(ed_cls):
-    import asynckivy as ak
-    eds = [ed_cls() for __ in range(3)]
-    async def _test():
-        tasks = await ak.or_(*(ak.event(ed, 'on_test') for ed in eds))
-        assert not tasks[0].done
-        assert tasks[1].done
-        assert not tasks[2].done
-        nonlocal done;done = True
-    done = False
-    ak.start(_test())
-    assert not done
-    eds[1].dispatch('on_test')
-    assert done
+class Test_or_:
+    def test_normal(self, ed_cls):
+        import asynckivy as ak
+        eds = [ed_cls() for __ in range(3)]
+        async def _test():
+            tasks = await ak.or_(*(ak.event(ed, 'on_test') for ed in eds))
+            assert not tasks[0].done
+            assert tasks[1].done
+            assert not tasks[2].done
+            nonlocal done;done = True
+        done = False
+        ak.start(_test())
+        assert not done
+        eds[1].dispatch('on_test')
+        assert done
+
+    @pytest.mark.parametrize("n_do_nothing", range(1, 4))
+    def test_some_coroutines_immediately_end(self, n_do_nothing):
+        '''github issue #3'''
+        import asynckivy as ak
+        async def do_nothing():
+            pass
+        async def _test():
+            tasks = await ak.or_(
+                *(do_nothing() for __ in range(n_do_nothing)),
+                *(ak.sleep_forever() for __ in range(3 - n_do_nothing)),
+            )
+            for task in tasks[:n_do_nothing]:
+                assert task.done
+            for task in tasks[n_do_nothing:]:
+                assert not task.done
+            nonlocal done; done = True
+        done = False
+        ak.start(_test())
+        assert done
 
 
-def test_and_(ed_cls):
-    import asynckivy as ak
-    eds = [ed_cls() for __ in range(3)]
-    async def _test():
-        tasks = await ak.and_(*(ak.event(ed, 'on_test') for ed in eds))
-        assert tasks[0].done
-        assert tasks[1].done
-        assert tasks[2].done
-        nonlocal done;done = True
-    done = False
-    ak.start(_test())
-    assert not done
-    eds[1].dispatch('on_test')
-    assert not done
-    eds[0].dispatch('on_test')
-    assert not done
-    eds[2].dispatch('on_test')
-    assert done
+class Test_and_:
+    def test_normal(self, ed_cls):
+        import asynckivy as ak
+        eds = [ed_cls() for __ in range(3)]
+        async def _test():
+            tasks = await ak.and_(*(ak.event(ed, 'on_test') for ed in eds))
+            assert tasks[0].done
+            assert tasks[1].done
+            assert tasks[2].done
+            nonlocal done;done = True
+        done = False
+        ak.start(_test())
+        assert not done
+        eds[1].dispatch('on_test')
+        assert not done
+        eds[0].dispatch('on_test')
+        assert not done
+        eds[2].dispatch('on_test')
+        assert done
+
+    @pytest.mark.parametrize("n_coros", range(1, 4))
+    def test_all_coroutines_immediately_end(self, n_coros):
+        '''github issue #3'''
+        import asynckivy as ak
+        async def do_nothing():
+            pass
+        async def _test():
+            tasks = await ak.and_(*(do_nothing() for __ in range(n_coros)))
+            for task in tasks:
+                assert task.done
+            nonlocal done; done = True
+        done = False
+        ak.start(_test())
+        assert done
 
 
 class TestEvent:

